src/promptflow-core/promptflow/core/_prompty_utils.py

```diff
@@ -128,14 +135,13 @@ def convert_prompt_template(template, inputs, api):
 
     # convert list type into ChatInputList type
     converted_kwargs = convert_to_chat_list(inputs)
-    rendered_prompt = render_jinja_template_content(
-        template_content=prompt, trim_blocks=True, keep_trailing_newline=True, **converted_kwargs
-    )
     if api == "completion":
-        return rendered_prompt
+        rendered_prompt = render_jinja_template_content(
+            template_content=prompt, trim_blocks=True, keep_trailing_newline=True, **converted_kwargs
+        )
     else:
-        referenced_images = find_referenced_image_set(inputs)
-        return parse_chat(rendered_prompt, list(referenced_images))
+        rendered_prompt = build_messages(prompt=prompt, **converted_kwargs)
```

Comment:
please run a e2e test in local and add a screenshot of trace like in the tool escape PR descriptions

Reply:
Add screenshot to description